proposals/gc/MVP.md

```diff
@@ -379,29 +387,27 @@ TODO: Add the ability to generate new (non-canonical) RTT values to implement ca
 
 RTT-based casts can only be performed with respect to concrete types, and require a data or function reference as input, which are known to carry an RTT.
 
-* `ref.test <typeidx>` tests whether a reference value's [runtime type](#values) is a [runtime subtype](#runtime) of a given RTT
-  - `ref.test $t : [(ref null ht) (rtt n? $t)] -> [i32]`
-    - iff `ht <: data` or `ht <: func`
-    - and `(type $t) <: ht`
+* `ref.test` tests whether a reference value's [runtime type](#values) is a [runtime subtype](#runtime) of a given RTT
+  - `ref.test : [t' (rtt n? $t)] -> [i32]`
+    - iff `t' <: dataref` or `t' <: funcref`
```

Comment:
Ah, no, the purpose of introducing the separate classification instructions actually was to simplify the test/cast instructions, so that they only operate on references that are known to have an RTT. Otherwise there function would overlap with the classification instructions. See #150 for more context.

You are right about the null thing, fixed.